### ISSUE TITLE
Implement and test explicit coupling

### ISSUE THREAD

**Issue by BenjaminRueth** [precice/fenics-adapter#72]
To support explicit coupling, some things still have to be implemented. Generally, there are two flavours that we would like to support:

* FEniCS participant only writes data. See also https://github.com/precice/fenics-adapter/pull/59#discussion_r432432041
* FEniCS participant only reads data.

**Comment by BenjaminRueth** [comment]
In theory: Should work out of the box.
In practice: Needs to be tested!

**Comment by BenjaminRueth** [comment]
I did a quick test using the [partitioned heat example](https://github.com/precice/tutorials/tree/master/HT/partitioned-heat/fenics-fenics) with the following modified configuration file:

```
...
      <coupling-scheme:serial-explicit>
         <participants first="HeatDirichlet" second="HeatNeumann"/>
         <max-time value="1"/>
         <time-window-size value=".1" valid-digits="8"/>
         <exchange data="Flux"        mesh="NeumannNodes" from="HeatDirichlet" to="HeatNeumann" />
         <exchange data="Temperature" mesh="NeumannNodes" from="HeatNeumann"   to="HeatDirichlet" initialize="true"/>
      </coupling-scheme:serial-explicit>
...
```

I did not have to modify the adaper or `heat.py` (only comment out error checks) to run this case. The result looks like expected:

![Screenshot from 2020-10-19 16-21-01](https://user-images.githubusercontent.com/5740604/96464035-a4a17000-1227-11eb-841e-af480fc7fd97.png)

Due to explicit coupling the contour lines are not continuous and we do not  get the exact solution anymore.

I will keep this issue open, since the two point mentioned [above](https://github.com/precice/fenics-adapter/issues/72#issue-627213068) are still untested:

>     * FEniCS participant only writes data. See also [#59 (comment)](https://github.com/precice/fenics-adapter/pull/59#discussion_r432432041)
> 
>     * FEniCS participant only reads data.

**Comment by BenjaminRueth** [comment]
I quickly put together a one-way coupling variant of the partitioned heat example, where the Neumann solver only writes and the Dirichlet solver only receives (see [one-way.zip](https://github.com/precice/fenics-adapter/files/5403116/one-way.zip)).

Generally, it works. However, I observed an issue:

```
precice_dt = precice.initialize(coupling_boundary, mesh, V_g, write_function=u_D_function)
```

`initialize` always requires a function space. Even, if the participant does not read any data. However, for one-way coupling the function space should be optional, since it is not needed by the participant that writes.

I propose the following signature for `initialize`:

```
precice_dt = precice.initialize(coupling_boundary, mesh, read_function_space=V_g, write_function=u_D_function)
```

Related to #71.

**Comment by IshaanDesai** [comment]
> I propose the following signature for `initialize`:
> 
> ```
> precice_dt = precice.initialize(coupling_boundary, mesh, read_function_space=V_g, write_function=u_D_function)
> ```
> 
> Related to #71.

This is being handled in https://github.com/precice/fenics-adapter/pull/71 as having separate function spaces for read and write functions was already discussed there.

**Comment by BenjaminRueth** [comment]
Let's do a final verification of this functionality before merging #71 and then close this issue. No need for a unit test.

**Comment by BenjaminRueth** [comment]
Did another test of explicit and one-way coupling. [one-way.zip](https://github.com/precice/fenics-adapter/files/5720025/one-way.zip)

Everything works. I just found a small bug that is fixed in #107
